storage/store.go

```diff
@@ -1231,9 +1231,7 @@ func (s *Store) ExecuteCmd(ctx context.Context, args proto.Request) (proto.Respo
 			// clients will retry.
 			err = proto.NewRangeNotFoundError(rng.Desc().RangeID)
 		} else if iErr, ok := err.(*errWithIndex); ok {
-			err, index = iErr.err, iErr.index
-			// TODO(tschottdorf): propagate the index up. See #1891.
-			_ = index
+			err, index = iErr.err, &iErr.index
```

Comment:
Storing a pointer into the `errWithIndex` struct could have surprising results if that struct is later reused, and also prevents GC of the original error. Better to make a copy of `iErr.index` with `index = proto.Int32(iErr.index)`.